File: model/src/main/java/com/apeiron/igor/model/memory/NewGame.java
package com.apeiron.igor.model.memory;

import com.apeiron.igor.model.db.Person;
import com.apeiron.igor.model.db.User;
import com.apeiron.igor.model.memory.map.Map;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class NewGame {
    private String id;
    private Date createDate;
    private User enemyPlayer;
    private List<Person> personList;
    private Map gameMap;
}

File: web-client/src/main/js/igor-client/src/app/model/new-game.ts
import {User} from "./user";
import {Person} from "./person";
import {GameMap} from "./game-map";

export class NewGame {

  public id: string;
  public createDate: Date;
  public enemyPlayer: User;
  public personList: Person[];
  public gameMap: GameMap;
  public yourPersons: Person[];
  public enemyPlayerPersons: Person[];

  constructor(id?: string, createDate?: Date, enemyPlayer?: User, personList?: Person[], gameMap?: GameMap, yourPersons?: Person[], enemyPlayerPersons?: Person[]) {
    this.id = id;
    this.createDate = createDate;
    this.enemyPlayer = enemyPlayer;
    this.personList = personList;
    this.gameMap = gameMap;
    this.yourPersons = yourPersons;
    this.enemyPlayerPersons = enemyPlayerPersons;
  }

  public static fromJson(object:JSON):NewGame{
    let result:NewGame = new NewGame();
    result.id = object["id"];
    result.createDate = object["createDate"];
    result.enemyPlayer = User.fromJson(object["user"]);
    result.personList = Person.fromJsonArray(object["personList"]);
    result.gameMap = GameMap.fromJson(object["gameMap"]);
    result.yourPersons = Person.fromJsonArray(object["yourPersons"]);
    result.enemyPlayerPersons = Person.fromJsonArray(object["enemyPlayerPersons"]);
    return result;
  }
}
